pkg/jx/cmd/preview.go

```diff
@@ -236,6 +237,41 @@ func (o *PreviewOptions) Run() error {
 		label = envName
 	}
 
+	// we need pull request info to include
+	authConfigSvc, err := o.Factory.CreateGitAuthConfigService()
+	if err != nil {
+		return err
+	}
+
+	gitKind, err := o.GitServerKind(gitInfo)
+	if err != nil {
+		return err
+	}
+
+	gitProvider, err := gitInfo.CreateProvider(authConfigSvc, gitKind)
+
+	user := gitProvider.UserInfo(gitProvider.CurrentUsername())
+	prNum, err := strconv.Atoi(prName)
+	if err != nil {
+		log.Warn("Unable to convert PR " + prName + " to a number" + "\n")
+	}
+
+	pullRequest, _ := gitProvider.GetPullRequest(gitInfo.Organisation, gitInfo.Name, prNum)
+
+	statuses, err := gitProvider.ListCommitStatus(gitInfo.Organisation, gitInfo.Name, pullRequest.LastCommitSha)
+
+	if err != nil {
+		log.Warn("Unable to get statuses for PR " + prName + "\n")
+	}
+
+	buildStatus := ""
+	buildStatusUrl := ""
+	if len(statuses) > 0 {
+		status := statuses[len(statuses)-1]
+		buildStatus = status.State
```

Comment:
It's unclear what to do here, none of the data returned _seems_ to be a reasonable way to actually identify the status (E.g. "failed" "building" etc.)

Reply:
each URL has its own status as folks could add a jenkins + sonarqube check for example and each one can be triggered N times  etc. All we can know is that for each unique URL they all pass or one of them fails

Reply:
@jstrachan it's actually more complicated than that. we could retrigger a build manually due to some environmental failure, which results in success but the old statuses remain in the history, so it would always show failure despite being passing if we did that